lib/provisioning/Provisioner.js

```diff
@@ -2,13 +2,17 @@
 "use strict";
 var Promise = require("bluebird");
 var IrcRoom = require("../models/IrcRoom");
+var IrcAction = require("../models/IrcAction");
 var MatrixRoom = require("matrix-appservice-bridge").MatrixRoom;
+var StateLookup = require('matrix-appservice-bridge').StateLookup;
 
 var log = require("../logging").get("Provisioner");
+var promiseutil = require("../promiseutil.js");
 
 function Provisioner(ircBridge, enabled) {
     this._ircBridge = ircBridge;
     this._enabled = enabled;
+    this._expectations = [];
```

Comment:
If feels like the wrong data structure here. What you want is surely:
```javascript
{
  $domain: {
    $nick: {
      defer: Deferred
    }
  }
}
```
AKA a map of IRC users to Deferreds, where the Deferreds representing a pending auth request... so maybe `nicksToAuthRequests` or something?